maitake/src/wait/cell.rs

```diff
@@ -103,23 +141,25 @@ impl WaitCell {
                     "state changed unexpectedly while parking!"
                 );
 
-                if let Some(prev_waker) = prev_waker {
-                    prev_waker.wake();
-                }
-
                 if let Some(waker) = waker {
                     waker.wake();
                 }
 
-                if test_dbg!(state.is(State::CLOSED)) {
-                    wait::closed()
-                } else {
-                    wait::notified(())
-                }
+                // We just went to the closed state, so sorry new waker, shops
+                // closed
+                closed()
             }
         }
     }
 
+    /// Wait to be woken up by this cell.
+    pub fn wait(&self) -> Wait<'_> {
+        Wait {
+            cell: self,
+            registered: false,
+        }
+    }
+
     pub fn notify(&self) -> bool {
```

Comment:
this is completely unrelated to this change, but i think we should probably rename this method to `wake`, for consistency with the other primitives in `mycelium_util::wake`. we can do that in a separate PR, but it was just something i noticed while i was here.